android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginViewModel.kt

```diff
@@ -120,25 +122,21 @@ constructor(
     val locationHierarchies =
       practitionerDetails.fhirPractitionerDetails.locationHierarchyList ?: listOf()
 
-    val careTeamIds = fhirEngine.create(*careTeams.toTypedArray())
-    val organizationIds = fhirEngine.create(*organizations.toTypedArray())
-    val locationIds = fhirEngine.create(*locations.toTypedArray())
+    val careTeamIds =
+      defaultRepository.create(*careTeams.toTypedArray()).map { it.getSubstringBetween("/", "/") }
+    val organizationIds =
+      defaultRepository.create(*organizations.toTypedArray()).map {
+        it.getSubstringBetween("/", "/")
+      }
+    val locationIds =
+      defaultRepository.create(*locations.toTypedArray()).map { it.getSubstringBetween("/", "/") }
 
+    sharedPreferences.write(SyncStrategy.PRACTITIONER.value, practitionerDetails.userDetail)
+    sharedPreferences.write(SyncStrategy.CARETEAM.value, careTeamIds)
+    sharedPreferences.write(SyncStrategy.ORGANIZATION.value, organizationIds)
+    sharedPreferences.write(SyncStrategy.LOCATION.value, locationIds)
     sharedPreferences.write(
-      SharedPreferenceKey.PRACTITIONER_DETAILS_USER_DETAIL.name,
-      practitionerDetails.userDetail
```

Comment:
I think we should be saving `practitionerDetails` here instead. The ` practitionerDetails.userDetail` is a `KeycloakUserDetails` which is a property of `PractitionerDetails`. Using `KeycloakUserDetails` id, we should be using the Practitioner ID instead.

I have fixed this in a separate branch.

Reply:
Sure, that's ok